advancedlog/color_converter.go

```diff
@@ -0,0 +1,68 @@
+package logger
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+const resetCode = "\u001b[0m"
+
+var ansiEscapeCodeToLevel = map[string]Level{
+	"\u001b[31;1m": ErrorLevel,
+	"\u001b[33;1m": WarnLevel,
+	"\u001b[34;1m": InfoLevel,
+	"\u001b[32;1m": DoneLevel,
+	"\u001b[35;1m": DebugLevel,
+}
+
+func convertColoredString(message string) (Level, string) {
```

Comment:
Would be good to comment what we consider a colored message ([from this comment](https://github.com/bitrise-io/go-utils/pull/166#discussion_r977686832)):
- starts with a color code
- can have trailing whitespace characters but the first non-whitespace characters have to be part of the reset-color code
- contains exactly one color and reset code pair

Reply:
updated in https://github.com/bitrise-io/go-utils/pull/166/commits/3d6a1120d6943e0fc7f55f6d67bc8eafa6f14e6e